game-core/src/main/java/games/strategy/triplea/delegate/EditDelegate.java

```diff
@@ -50,7 +50,7 @@ public String removeUnits(final Territory territory, final Collection<Unit> unit
     if (result != null) {
       return result;
     }
-    if (units == null || units.isEmpty()) {
+    if (units.isEmpty()) {
```

Comment:
Just wondering...  Why wouldn't we add a precondition at the top of this method if `units` being `null` is going to trip an NPE anyway?  I can see a potential runtime behavior change if `units` was `null` but `result` was non-`null` (i.e. we'd now be raising an NPE where it wouldn't have been raised in the past because there was an error entering edit mode, which would short circuit the NPE-triggering code).

But if someone's calling this method with an illegal value for `units`, should we not raise an NPE in all scenarios?